### PR TITLE
Get rid of statism

### PR DESCRIPTION
**Pull request by greg0ire**
That way you can have several instances with different settings (there
are no settings yet, but we could imagine introducing some in the future).

### CODE REVIEW

File: tests/SqlFormatterTest.php [hunk 1]
```diff
@@ -22,34 +22,38 @@ final class SqlFormatterTest extends TestCase
     /** @var string[] */
     private $sqlData;
 
+    /** @var SqlFormatter */
+    private static $formatter;
+
     public static function setUpBeforeClass() : void
     {
+        self::$formatter = new SqlFormatter();
```
Comment (goetas) [inline tests/SqlFormatterTest.php]: What about using `setUp()` instead of `setUpBeforeClass` ?

File: src/SqlFormatter.php [hunk 2]
```diff
@@ -748,144 +748,144 @@ final class SqlFormatter
...
+            'entries'=>count($this->tokenCache),
+            'size'=>strlen(serialize($this->tokenCache)),
         ];
     }
 
     /**
      * Stuff that only needs to be done once.  Builds regular expressions and sorts the reserved words.
      */
-    private static function init() : void
+    private function init() : void
```
Comment (goetas) [inline src/SqlFormatter.php]: I hope that getting rid of all this `static` does not hurt performance. I see that this method uses was meant to cache some regex expressions

File: tests/SqlFormatterTest.php (hunk 1 above)
Comment (greg0ire) [reply tests/SqlFormatterTest.php]: I did that initially, but then there is a test about cache stats that failed. We have to use the same instance for every test for it to stay green. Or rework that test, which can be done later (the goal of this PR is to remove statism from src, as a BC-break).

File: src/SqlFormatter.php (hunk 2 above)
Comment (greg0ire) [reply src/SqlFormatter.php]: I think it won't if people are careful to reuse the same instance, which is the case nowadays with DI.

**Comment by greg0ire** [comment]
There has been a conflict, please approve again @goetas

**Comment by greg0ire** [comment]
Thank  you!

**Pull request merged**